src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSchemas.cs

```diff
@@ -16,40 +17,41 @@ namespace System.Xml.Serialization
     using System.Security;
     using System.Net;
     using System.Reflection;
+    using System.Diagnostics.CodeAnalysis;
 
     public class XmlSchemas : CollectionBase, IEnumerable<XmlSchema>
     {
-        private XmlSchemaSet _schemaSet;
-        private Hashtable _references;
-        private SchemaObjectCache _cache; // cached schema top-level items
+        private XmlSchemaSet? _schemaSet;
+        private Hashtable? _references;
+        private SchemaObjectCache? _cache; // cached schema top-level items
         private bool _shareTypes;
-        private Hashtable _mergedSchemas;
+        private Hashtable? _mergedSchemas;
         internal Hashtable delayedSchemas = new Hashtable();
         private bool _isCompiled;
-        private static volatile XmlSchema s_xsd;
-        private static volatile XmlSchema s_xml;
+        private static volatile XmlSchema? s_xsd;
+        private static volatile XmlSchema? s_xml;
 
         public XmlSchema this[int index]
         {
-            get { return (XmlSchema)List[index]; }
+            get { return (XmlSchema)List[index]!; }
             set { List[index] = value; }
         }
 
-        public XmlSchema this[string ns]
+        public XmlSchema? this[string? ns]
         {
             get
             {
                 IList values = (IList)SchemaSet.Schemas(ns);
                 if (values.Count == 0)
                     return null;
                 if (values.Count == 1)
-                    return (XmlSchema)values[0];
+                    return (XmlSchema?)values[0];
 
                 throw new InvalidOperationException(SR.Format(SR.XmlSchemaDuplicateNamespace, ns));
             }
         }
 
-        public IList GetSchemas(string ns)
+        public IList GetSchemas(string? ns)
         {
             return (IList)SchemaSet.Schemas(ns);
         }
```

Comment:
`public int Add(XmlSchema schema, Uri baseUri)` will throw NRE on null schema and baseUri, also anything which calls Find will throw as it will call Preprocess which will throw NRE when passed null. I'm leaning toward leaving those all as non-null.

agreed on baseUri